test/runner/lib/integration/__init__.py

```diff
@@ -0,0 +1,214 @@
+
+        yield IntegrationEnvironment(integration_dir, inventory_name, ansible_config, vars_file)
+        return
+
+    root_temp_dir = os.path.expanduser('~/.ansible/test/tmp')
+
+    prefix = '%s-' % target.name
+    suffix = u'-\u00c5\u00d1\u015a\u00cc\u03b2\u0141\u00c8'
+
+    if args.no_temp_unicode or 'no/temp_unicode/' in target.aliases:
+        display.warning('Disabling unicode in the temp work dir is a temporary debugging feature that may be removed in the future without notice.')
+        suffix = '-ansible'
+
+    if isinstance('', bytes):
+        suffix = suffix.encode('utf-8')
+
+    if args.explain:
+        temp_dir = os.path.join(root_temp_dir, '%stemp%s' % (prefix, suffix))
+    else:
+        make_dirs(root_temp_dir)
+        temp_dir = tempfile.mkdtemp(prefix=prefix, suffix=suffix, dir=root_temp_dir)
+
+    try:
+        display.info('Preparing temporary directory: %s' % temp_dir, verbosity=2)
+
+        inventory_names = {
+            PosixIntegrationConfig: 'inventory',
+            WindowsIntegrationConfig: 'inventory.winrm',
+            NetworkIntegrationConfig: 'inventory.networking',
+        }
+
+        inventory_name = inventory_names[type(args)]
+
+        cache = IntegrationCache(args)
+
+        target_dependencies = sorted([target] + list(cache.dependency_map.get(target.name, set())))
+
+        files_needed = get_files_needed(target_dependencies)
+
+        integration_dir = os.path.join(temp_dir, 'test/integration')
+        ansible_config = os.path.join(integration_dir, '%s.cfg' % args.command)
+
+        file_copies = [
+            ('test/integration/%s.cfg' % args.command, ansible_config),
+            ('test/integration/integration_config.yml', os.path.join(integration_dir, vars_file)),
+            (inventory_path, os.path.join(integration_dir, inventory_name)),
+        ]
+
+        file_copies += [(path, os.path.join(temp_dir, path)) for path in files_needed]
+
+        directory_copies = [
+            (os.path.join('test/integration/targets', target.name), os.path.join(integration_dir, 'targets', target.name)) for target in target_dependencies
+        ]
+
+        inventory_dir = os.path.dirname(inventory_path)
+
+        host_vars_dir = os.path.join(inventory_dir, 'host_vars')
+        group_vars_dir = os.path.join(inventory_dir, 'group_vars')
+
+        if os.path.isdir(host_vars_dir):
+            directory_copies.append((host_vars_dir, os.path.join(integration_dir, os.path.basename(host_vars_dir))))
+
+        if os.path.isdir(group_vars_dir):
+            directory_copies.append((group_vars_dir, os.path.join(integration_dir, os.path.basename(group_vars_dir))))
+
+        directory_copies = sorted(set(directory_copies))
+        file_copies = sorted(set(file_copies))
+
+        if not args.explain:
+            make_dirs(integration_dir)
+
+        for dir_src, dir_dst in directory_copies:
+            display.info('Copying %s/ to %s/' % (dir_src, dir_dst), verbosity=2)
```

Comment:
Is a verbosity of 2 a bit too low for the amount of files potentially being copied?

Reply:
This only shows the directory names, not the files in them, so the list is short.

Usually between the file and directory copies there are maybe ~6 entries shown.

Reply:
Ok sounds good